### DISCUSSION TITLE
Problems converting JPEG from CMYK to RGB

### DISCUSSION THREAD

**Discussion by piquan** [ImageMagick/ImageMagick#3006] [Help]
I have a JPEG that's in CMYK space, and I want to convert it to sRGB.  But my attempts so far have failed.  Can anybody help me out here?

`identify -format %[colorspace] input.jpg` correctly identifies it as CMYK.  In `identify -verbose input.jpg`, I don't see anything about the profile, but from other context I think it's SWOP v2.  I want to emit sRGB.

`display input.jpg` shows the image as it should appear.  Unsurprisingly, things like my filesystem browser aren't equipped to handle CMYK JPEGs; they display the image incorrectly, like a negative.

Here are some of the commands I've tried:
```
$ convert input.jpg -colorspace sRGB output.jpg
$ convert input.jpg -profile /usr/share/color/icc/sRGB.icc output.jpg
$ convert input.jpg -profile /usr/share/color/icc/SWOP2006_Coated3v2.icc -profile /usr/share/color/icc/sRGB.icc output.jpg
$ convert input.jpg -strip -profile /usr/share/color/icc/SWOP2006_Coated3v2.icc -profile /usr/share/color/icc/sRGB.icc output.jpg
# I also try forcing convert to alter the picture to make sure there aren't back-channel hijinks going on.
$ convert input.jpg -profile /usr/share/color/icc/SWOP2006_Coated3v2.icc -profile /usr/share/color/icc/sRGB.icc -convolve 0,0,0,0,1,0,0,0,0 output.jpg
$ convert input.jpg intermediate.png ; convert intermediate.png output.jpg
$ convert input.jpg -profile /usr/share/color/icc/SWOP2006_Coated3v2.icc intermediate.png ; convert intermediate.png output.jpg
```

Different variations give different results; sometimes I'll get something nonsensical like an image tagged as being a CMYK image with the sRGB colorspace.  But in all cases, I get something that isn't RGB: things like my filesystem browser (and usually `display`) still show the negative, as if the conversion just put the C, M, and Y channels into the R, G, and B channels without actually doing the colorspace conversions.

```
$ convert -version
Version: ImageMagick 6.9.11-24 Q16 x86_64 20200718 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib
```

**Comment by snibgo** [comment]
Please link to the file. If you can't do that, paste the output from "identify -verbose input.jpg".

Do you have problems with other files, or just this one? Perhaps this file is bad, and contains sRGB pixel values but metadata that claims it is CMYK. Try:
```
convert input.jpg -set colorspace sRGB x.jpg
```
That shouldn't work. That command will change metadata to claim the file is sRGB. If your input.jpg is good, this will invert the colors that make it look wrong. But if the file is bad, then this will correct the problem so they will no longer look inverted.
